src/core/thread/mle.cpp

```diff
@@ -487,6 +487,11 @@ ThreadError Mle::SetMeshLocalPrefix(const uint8_t *aMeshLocalPrefix)
     // Add the address back into the table.
     mNetif.AddUnicastAddress(mMeshLocal64);
 
+    if (mDeviceState != kDeviceStateDetached)
```

Comment:
Also need to exclude `kDeviceStateDisabled`.  How about checking `mDeviceState >= kDeviceStateChild`?